### PR TITLE
Fixes #29148 - Use Puma instead of Passenger by default

### PR DESCRIPTION
**Pull request by sthirugn**

### CODE REVIEW

**Comment by ekohl** [comment]
Two nits about the PR title: the `#` is significant (note that under the `Checks` tab no issue was linked). I'd also say what you are doing, so something like `Use Puma instead of Passenger by default`.

**Comment by sthirugn** [comment]
**Test results**
I was able to test locally using the following vagrant configuration.  foreman-service came up successfully with the required puma processes.

**Vagrant box definition**
```yaml
centos7-puma-default:
  box: centos7-katello-nightly
  memory: 12000
  ansible:
    variables: # defaults in roles/hammer_devel/defaults/main.yml
      hammer_devel_github_username: blah
      foreman_repositories_environment: staging
      foreman_installer_options:
        - "--puppet-server-jvm-min-heap-size 300M"
        - "--puppet-server-jvm-max-heap-size 300M"
      foreman_installer_module_prs:
        - theforeman/puppet-foreman/802
```

**Comment by sthirugn** [comment]
Updated spec tests

**Comment by sthirugn** [comment]
Updated more spec tests and squashed all the commits. Ready for the final review once the tests pass.

File: spec/classes/foreman_spec.rb [hunk 1]
```diff
@@ -145,22 +145,26 @@
...
+      let(:params) { super().merge(passenger: true) }
+
+      it do
+        should contain_exec('restart_foreman')
+          .with_command('/bin/touch /usr/share/foreman/tmp/restart.txt')
+          .with_refreshonly(true)
+          .with_cwd('/usr/share/foreman')
+          .with_path('/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin')
+      end
+    end
```
Comment (ehelms) [inline spec/classes/foreman_spec.rb]: I think the indentation is off by a level.

**Comment by ehelms** [comment]
Looks good, I think the tests failing are same as master and this is good to go in for nightly to begin using puma.

**Comment by ekohl** [comment]
Currently no migration has been written to switch existing installations. IMHO that's fine and I'd feel comfortable not trying to migrate everyone in 2.1, leaving just fresh installations. Just wanted to make sure we're all on the same page about that rather than an omission.

**Comment by ehelms** [comment]
Yep, the migration is in progress. I prefer to upgrade users to the same as
freah installs to reduce matrix of deployment types and allow hardening of
puma.

**Comment by sthirugn** [comment]
Tested this change in stage and it worked great:
```sh
# journalctl -u foreman -f
-- Logs begin at Mon 2020-03-02 17:45:39 UTC. --
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] Puma starting in cluster mode...
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] * Version 3.11.4 (ruby 2.5.5-p157), codename: Love Song
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] * Min threads: 0, max threads: 16
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] * Environment: production
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] * Process workers: 2
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] * Phased restart available
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] * Listening on tcp://127.0.0.1:3000
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] Use Ctrl-C to stop
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] - Worker 0 (pid: 17647) booted, phase: 0
Mar 02 17:59:55 centos7-puma-staging-test.apex.example.com foreman[17405]: [17464] - Worker 1 (pid: 17656) booted, phase: 0
```

Box definition:
```sh
centos7-puma-staging-test:
  box: centos7-katello-nightly
  memory: 12000
  ansible:
    variables: # defaults in roles/hammer_devel/defaults/main.yml
      hammer_devel_github_username: *****
      foreman_repositories_environment: staging
      foreman_installer_options:
        - "--puppet-server-jvm-min-heap-size 300M"
        - "--puppet-server-jvm-max-heap-size 300M"
```

**Pull request merged**